lib/task.js

```diff
@@ -140,8 +140,19 @@ export default class Task {
         this.task.payload, this.status.taskId
     );
 
+    let taskCapabilities = this.task.payload.capabilities || {};
+    let privilegedTask = taskCapabilities.privileged || false;
+    if (privilegedTask && !this.runtime.dockerConfig.allowPrivileged) {
```

Comment:
I think this is solid... An alternative and/or additional way of restricting this is to require a scope for privileged tasks.
Ie. require `docker-worker:privileged-container` to be covered by `task.scopes` (they same thing we do for cache folders).

The scope could also be: `docker-worker:privileged-container:<provisionerId>/<workerType>`, since that would make it easier to issue tighter controlled scopes.

Anyways, I like that it can be completely disabled for a given workerType :)